src/examples/utilities/components.ts

```diff
@@ -0,0 +1,172 @@
+- \`'down'\` match screen widths included within the given breakpoint and down
+- \`'only'\` match screen widths included within the given breakpoint
+- \`'between'\` match screen widths including the first breakpoint up through screen widths included within the second breakpoint`
+      },
+      breakpoint: {
+        required: true,
+        type: "'xs' | 'sm' | 'md' | 'lg' | 'xl' | Array",
+        description:
+          "Specifies a theme object [breakpoints](/components/theme-object#breakpoints) key, or an array of two keys when`'between'` is the specified query"
+      },
+      theme: {
+        type: 'DefaultTheme',
+        defaultValue: 'DEFAULT_THEME',
+        description: 'Provides values used to resolve the specified breakpoint'
+      }
+    }
+  },
+  {
+    name: 'menuStyles',
+    props: {
+      position: {
+        required: true,
+        type: "'top' | 'right' | 'bottom' | 'left'",
+        description:
+          'Specifies the menu position relative to the component that triggers menu expansion'
+      },
+      '[options.theme]': {
+        type: 'DefaultTheme',
+        defaultValue: 'DEFAULT_THEME',
+        description: 'Provides theming values used to style the menu component'
+      },
+      '[options.hidden]': {
+        type: 'boolean',
+        description: 'Determines whether the menu is hidden'
+      },
+      '[options.margin]': {
+        type: 'string',
+        description:
+          'Determines the amount of space, as a CSS dimension, between the menu and its trigger'
+      },
+      '[options.zIndex]': {
+        type: 'number',
+        defaultValue: 0,
+        description: 'Specifies the `z-index` for the absolutely positioned menu wrapper'
+      },
+      '[options.childSelector]': {
+        type: 'string',
+        defaultValue: "'> *'",
+        description:
+          'Indicates a CSS selector which, when applied to the wrapper, identifies the child menu component'
+      },
+      '[options.animationModifier]': {
+        type: 'string',
+        description:
+          "Indicates a CSS class or attribute selector which, when applied to the wrapper, animates the menu's appearance"
+      }
+    }
+  },
+  {
+    name: 'retrieveComponentStyles',
+    props: {
+      id: {
+        required: true,
+        type: 'string',
+        description:
+          "Specifies the unique ID used to retrieve CSS styles from the theme's [components](/components/theme-object#components) object"
+      },
+      props: {
+        required: true,
+        type: 'Partial<ThemeProps<Partial<DefaultTheme>>>',
+        description: 'Provides component props which contain the context `theme` object'
+      }
+    }
+  },
+  {
+    name: 'withTheme',
+    props: {
+      Component: {
+        required: true,
+        type: 'Object',
+        description:
+          'Takes a component to be wrapped by a higher-order component with an added `theme` prop'
```

Comment:
This one is kind of hard to parse. What does it mean to "take a component"?

Reply:
@vsrixyz I agree. Yet this is parlance that is familiar in React land. Check out https://reactjs.org/docs/higher-order-components.html

Reply:
...although, I think I can do better.

> Gets wrapped by a higher-order component with an added `theme` prop